compiler/rustc_resolve/src/lib.rs

```diff
@@ -1584,6 +1586,57 @@ impl<'a> Resolver<'a> {
         });
     }
 
+    fn check_reexport_ambiguities(&mut self) {
+        for module in self.arenas.local_modules().iter() {
+            if let Some(_) = module.opt_def_id() {
+                let mut ambiguous_reexports = FxHashMap::default();
+
+                module.for_each_child(self, |this, ident, ns, binding| {
+                    if let NameBindingKind::Import { import: reexport, .. } = binding.kind
+                    && let res = binding.res().expect_non_local::<Res>()
+                    && res != def::Res::Err {
+                        if let Some((ambiguous_binding, _)) = binding.ambiguity
+                        && let NameBindingKind::Import { import: ambiguous_reexport, .. } = ambiguous_binding.kind {
```

Comment:
This condition and conditions below do not make sense to me.
If `binding.ambiguity.is_some()` and `is_exported(import.id())` then you have something to report.